api/applications_api/serializers.py

```diff
@@ -298,6 +315,14 @@ class Meta:
             },
         }
 
+    def get_purpose_name(self, obj):
```

Comment:
Not 100% related to this PR, but purpose name doesn't seem to be unique, which it most likely should be. While this is read only thingy, i think when used like this should be maybe used only on unique values to be super clear. 

But no need to do anything in this pr, more or less for discussion.

Reply:
Yes, I think the purposes should be perhaps be pre defined => it could be just enum also

Reply:
Not sure if it can be an enum, might need some flexibility. Anyway this looks fine to me.